src/gsi/prewgt_reg.f90

```diff
@@ -289,6 +475,135 @@ subroutine prewgt_reg(mype)
               end do
            end do
         end do
+     else if ( n==nrf3_qr .or. n==nrf3_qs .or. n==nrf3_qg .or. n==nrf3_qnr) then
+
+        if ( l_be_T_dep ) then
+            if (mype==0) &
+                write(6,*)' prewgt_reg(mype=',mype,')',' Temperature-dependent erro vars for Q_cld and as3d= ',as3d(n),' for var:',cvars3d(n)
+            loc=nrf3_loc(n)
+
+!           re-define vz_cld on model sigma grid for cloud variabels (not on stats grid)
+!           to match the re-defined corz_cld, which is also defined on model grid
+!           because the error variables of cloud variable is modle/background temperature
+!           dependent.
+!           only initialization of alv in subroutine smoothzo
+            do j=llmin,llmax
+                call smoothzo(vz(1,j,n),samp,rate,n,j,dsv(1,1,j))
+            end do
+
+            do j=1,lat2
+                mid_nsig = INT((nsig+1)/2)
+                mid_mlat = INT((llmin+llmax)/2)
+                vz_cld(1:nsig) = vz(1:nsig,mid_mlat,n)    !
+                call smoothzo1(vz_cld(1:nsig),samp,rate,dsv_cld(1:lon2,1:nsig,j))
+                do i=1,lon2
+                    do k=1,nsig
+                        tsen=ges_tsen(j,i,k,nfldsig)
+                        if (n==nrf3_qr) then
+                            if( l_use_cvpqx==.true. .and. l_use_cvpqx_pval==0.5_r_kind) then
```

Comment:
1. I changed the code according to your comments (use ‘if(l_use_cvpqx) then’ ).
2. We also renamed ‘l_use_cvpqx_pval' as ‘cvpqx_pval’ to avoid misunderstanding. cvpqx_pval is real variable. 

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-e3997d5227cdf84b5d4eb757ede26e41eea18c533d66c52017f2008bbf32a660R500-R503

You can also see your updates here.
read_l2bufr_mod.f90
https://github.com/NOAA-EMC/GSI/pull/142/files#diff-f20faba007ef4692848f3d4f8945f0a6cd2dfe3fbfdaec6593c8a10d2d8b3719
read_radar.f90
https://github.com/NOAA-EMC/GSI/pull/142/files#diff-9237af61f8a4787143a6ef94bfdefcd7a764f8f986e54370beffb92b3f08d2d3R3833-R3839